src/MCPClient/lib/clientScripts/upload-archivesspace.py

```diff
@@ -190,36 +191,40 @@ def upload_to_archivesspace(files, client, xlink_show, xlink_actuate, object_typ
-        metavar="host", help="hostname of ArchivesSpace")
-    parser.add_argument('--port', type=int, default=8089, dest='port',
-        metavar="port", help="Port used by ArchivesSpace backend API")
-    parser.add_argument('--user', dest='user', metavar="Administrative user")
-    parser.add_argument('--passwd', dest='passwd', metavar="Administrative user password")
-    parser.add_argument('--dip_location', metavar="dip location")
-    parser.add_argument('--dip_name', metavar="dip name")
-    parser.add_argument('--dip_uuid', metavar="dip uuid")
-    parser.add_argument('--restrictions', metavar="restrictions apply", default="premis", choices=RESTRICTIONS_CHOICES)
-    parser.add_argument('--object_type', metavar="object type", default="")
-    parser.add_argument('--xlink_actuate', metavar="xlink actuate", default="onRequest", choices=EAD_ACTUATE_CHOICES)
-    parser.add_argument('--xlink_show', metavar="xlink show", default="new", choices=EAD_SHOW_CHOICES)
-    parser.add_argument('--use_statement', metavar="use statement")
-    parser.add_argument('--uri_prefix', metavar="uri prefix")
-    parser.add_argument('--access_conditions', metavar="conditions governing access", default="")
-    parser.add_argument('--use_conditions', metavar="conditions governing use", default="")
+    INHERIT_NOTES_CHOICES = ['yes', 'y', 'true', '1']
+
+    parser = argparse.ArgumentParser(description='A program to take digital objects from a DIP and upload them to an ArchivesSpace db')
+    parser.add_argument('--host', default='localhost', dest='host', metavar='host', help='Hostname of ArchivesSpace')
+    parser.add_argument('--port', type=int, default=8089, dest='port', metavar='port', help='Port used by ArchivesSpace backend API')
+    parser.add_argument('--user', dest='user', help='Administrative user')
+    parser.add_argument('--passwd', dest='passwd', help='Administrative user password')
+    parser.add_argument('--dip_location', help='DIP location')
+    parser.add_argument('--dip_name', help='DIP name')
+    parser.add_argument('--dip_uuid', help='DIP UUID')
+    parser.add_argument('--restrictions', help='Restrictions apply', default='premis', choices=RESTRICTIONS_CHOICES)
+    parser.add_argument('--object_type', help='object type', default='')
+    parser.add_argument('--xlink_actuate', help='XLink actuate', default='onRequest', choices=EAD_ACTUATE_CHOICES)
+    parser.add_argument('--xlink_show', help='XLink show', default='new', choices=EAD_SHOW_CHOICES)
+    parser.add_argument('--use_statement', help='USE statement')
+    parser.add_argument('--uri_prefix', help='URI prefix')
+    parser.add_argument('--access_conditions', help='Conditions governing access', default='')
+    parser.add_argument('--use_conditions', help='Conditions governing use', default='')
+    parser.add_argument('--inherit_notes', help='Inherit digital object notes from the parent component', default=False, type=str.lower, choices=INHERIT_NOTES_CHOICES)
```

Comment:
I like this solution much better than the previous one!

Nitpick: `default=False`, but `type=str` - the `default` should be the same type as the `type`.  I'd also move the `str.lower()` to below, eg `args.inherit_notes = args.inherit_notes.lower() in INHERIT_NOTES_CHOICES` instead of having it in the type.